open_spiel/python/mfg/algorithms/distribution.py

```diff
@@ -87,10 +88,18 @@ def evaluate(self):
             pyspiel.StateType.MEAN_FIELD):
         for mfg_state in listing_states:
           dist_to_register = mfg_state.distribution_support()
-          dist = list(map(self.value_str, dist_to_register))
-          assert (sum(dist) - len(self._root_states)) < 1e-4, (
-            "Sum of probabilities of all possible states should be the number of "
-            f"population, it is {sum(dist)}.")
+          def get_probability_for_state(str_state):
```

Comment:
Ah, I did not realize that we needed to fallback to a default 0 value here, but indeed we do.
I would not use exceptions for that, since it's not something exceptional (and catching exception can be slow).
I would call directly self.distribution.get(str_state, 0.) here.

See also:
https://google.github.io/styleguide/pyguide.html#24-exceptions

Reply:
THanks

Reply:
I would have to update https://github.com/deepmind/open_spiel/blob/104deea0f931f960b362a38e498410385ac5e81b/open_spiel/python/mfg/algorithms/best_response_value.py#L79